### ISSUE TITLE
Advice

### ISSUE THREAD

**Issue by Gingerlief** [MaxRev-Dev/gdal.netcore#134]
Im trying to build the nugets with MRSID sdk,(windows first) adding the compile routine to partials.ps1 and grabbing the required file from https://bin.extensis.com/download/developer/MrSID_DSDK-9.5.5.5244-win64-vc17.zip. 

My question is outside of the cmake portion of gdal to add the sdk, where would other changes be made to enable an external driver. 

Currently I have successfully gotten it to fully build on the v3.9.0 and function for .net 8.0, but it won't see the MRSID repo with GDAL.Open(...)

Any advice would be great. 

Thanks

**Comment by MaxRev-Dev** [comment]
You need to check the dependencies of [gdal.dll](https://github.com/MaxRev-Dev/gdal.netcore/tree/main/win#troubleshooting-dependencies)
I'm pretty sure that you have not copied the DLLs from MrSID's `bin` directory. 
CMake on windows doesn't do this automatically, at least with current implementation.

**Comment by Gingerlief** [comment]
$webpRoot = Get-ForceResolvePath("$env:BUILD_ROOT\sdk\libwebp*")
$env:WEBP_ROOT = "-DWEBP_INCLUDE_DIR=$webpRoot\include"
$env:WEBP_LIB = "-DWEBP_LIBRARY=$webpRoot\lib\libwebp.lib"
# MRSID setup, mimic webpRoot
$mrsidRoot = Get-ForceResolvePath("$env:BUILD_ROOT\sdk\Raster_DSDK")
$env:MRSID_ROOT = "-DMRSID_INCLUDE_DIR=$mrsidRoot\include"
$env:MRSID_LIB = "-DMRSID_LIBRARY=$mrsidRoot\lib\lti_dsdk.lib"
Write-BuildStep "Configuring GDAL"
Set-Location "$env:BUILD_ROOT"

Currently  "$env:MRSID_ROOT $env:MRSID_LIB  " was within the cmake list, are you saying that I should add in the DCMAKE_PREFIX_PATH add something like $mrsidRoot\bin ?

**Comment by MaxRev-Dev** [comment]
If GDAL compiles successfully and you see the library was recognized during GDAL configuration by CMake (like [here](https://github.com/MaxRev-Dev/gdal.netcore/actions/runs/9509495865/job/26230058347#step:6:1008)). Then the driver was liked into `gdal.dll`. Now you need to copy the DLL.

This not applies to WEBP library as it's linked statically. But by default all drivers are linked dynamically, so you need to copy the DLL after GDAL compiles. It's being done [here](https://github.com/MaxRev-Dev/gdal.netcore/blob/d30f0a28417ec7ca8d0b2058d3ebd3506cf5601b/win/collect-deps-makefile.vc#L32)

**Comment by Gingerlief** [comment]
I think I messed up there then. 
	$(COPY)  "$(abspath $(MRSID_ROOT)\bin)\*.dll" "$(OUTPUT)\" 
	$(COPY)  "$(abspath $(MRSID_ROOT)\lib)\*.dll" "$(OUTPUT)\" 
I had both bin+lib setup earlier because I read something similar, I will build with just the \bin as I haven't tried that.

**Comment by Gingerlief** [comment]
So I guess thats why I was doing the /lib, LTI_DSDK_9.5.dll doesn't exist within the /bin of raster_dsdk, and only exists in /lib
ldd is much easier than depends.exe 
![image](https://github.com/MaxRev-Dev/gdal.netcore/assets/11828947/377fe47c-0787-4251-8853-57699852bba1)

**Comment by MaxRev-Dev** [comment]
Yes. It makes sense now. Not sure why it's there as that's not a standard location. 
Anyways, once you copy it, it should solve the issue. You need to do this before the nuget pack executes.
The second line that you had in `collect-deps` should do the thing.

**Comment by Gingerlief** [comment]
So back to square 1, but I think I can see part of the problem. It looks like when I was copying over mrsid_root/bin it contains its own geos_c.dll/geos.dll, which is a dependency for spatialite and few others I assume. In turn I'm thinking I was corrupting it with a version mismatch or something.

**Comment by Gingerlief** [comment]
That Fixed it
